### ISSUE TITLE
ModuleNotFoundError: No module named 'imp' when installing with pip

### ISSUE THREAD

**Issue by pepal00** [droope/droopescan#77]
Hello,
I installed droopescan with pip inside a venv. When tried to run it I get the error `ModuleNotFoundError: No module named 'imp'`. I tried to install pystache==0.5.4 as recommended in [this comment issue #72](https://github.com/SamJoan/droopescan/issues/72#issuecomment-2145103726) but that installation errors with
```pystache: using: version '75.6.0' of <module 'setuptools' from '/tmp/pip-build-env-cd073l4n/overlay/lib/python3.12/site-packages/setuptools/__init__.py'>
      Warning: 'classifiers' should be a list, got type 'tuple'
      error in pystache setup command: use_2to3 is invalid.
```
Digging a bit on that, it seems like Python 3.12 has removed support for the imp library, so I think the module will need replaced with [importlib](https://docs.python.org/3.11/library/imp.html)?

* OS & installation method
Kali 6.11.2-1kali1 (2024-10-15) rolling, installed via pip
Python 3.12.7

**Comment by Chepycou** [comment]
Same issue and conclusion here

**Comment by cinvokesec** [comment]
I fixed this by bumping the version of `cement` up and adding `setuptools` in `setup.py`:

```
install_requires=[
        'cement>=2.10,<2.10.99',
        'setuptools',
        'requests',
        'pystache',
    ],
```

Then a pip install requirements and it worked.

The version currently used (2.6) has `imp` which is where my error was coming from. The [fix](https://github.com/datafolklabs/cement/issues/737) happened in 2.10. As a note, I did try v3 but it had errors and I didn't want to deal with those so I stuck with 2.10.14.

**Comment by GoogTech** [comment]
> Hello, I installed droopescan with pip inside a venv. When tried to run it I get the error `ModuleNotFoundError: No module named 'imp'`. I tried to install pystache==0.5.4 as recommended in [this comment issue #72](https://github.com/SamJoan/droopescan/issues/72#issuecomment-2145103726) but that installation errors with
> 
> ```
>       Warning: 'classifiers' should be a list, got type 'tuple'
>       error in pystache setup command: use_2to3 is invalid.
> ```
> 
> Digging a bit on that, it seems like Python 3.12 has removed support for the imp library, so I think the module will need replaced with [importlib](https://docs.python.org/3.11/library/imp.html)?
> 
> * OS & installation method
>   Kali 6.11.2-1kali1 (2024-10-15) rolling, installed via pip
>   Python 3.12.7

Yes, Python 3.12+ does not support the 'imp'. It works in the range of Python 3.8 ~ 3.11, maybe.

**Comment by Aggelos11** [comment]
Any solution?

**Comment by lunemoonfang** [comment]
pipx inject droopescan cement==2.10.14 setuptools --force

**Comment by abb00717** [comment]
My solution is 

```
$ uv python install 3.11
$ uv python find 3.11
<path to 3.11>

$ pipx install --force --python <path to 3.11> droopescan 2>&1
```

**Comment by nathan-lalli** [comment]
> pipx inject droopescan cement==2.10.14 setuptools --force

This worked best for me
